app/models/participation.rb

```diff
@@ -92,9 +95,9 @@ def set_default_notifications_flags
     self.send_reminder_notification = rdv.motif.visible_and_notified? if send_reminder_notification.nil?
   end
 
-  def new_raw_invitation_token
-    invite! { |rdv_u| rdv_u.skip_invitation = true }
-    raw_invitation_token
+  def set_restricted_authentication_token
+    self.restricted_auth_token = SecureRandom.send(:choose, [*"A".."Z", *"0".."9"], 8)
+    self.restricted_auth_token = SecureRandom.send(:choose, [*"A".."Z", *"0".."9"], 8) while Participation.where(restricted_auth_token:).any?
```

Comment:
nitpick pour le plaisir mais j’ai essayé de réécrire ça en plus DRY, c’est pas forcément plus clair 🤷 

```suggestion
    self.restricted_auth_token = SecureRandom.send(:choose, [*"A".."Z", *"0".."9"], 8) until restricted_auth_token && Participation.where(restricted_auth_token:).none?
```

Reply:
haha, je me disais bien qu'il y avait une manière plus jolie d'écrire ça. Merci !